src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -2951,6 +2952,52 @@ def get_pipeline(
 
             return pipeline.to_model(hydrate=hydrate)
 
+    def list_pipeline_namespaces(
+        self,
+        filter_model: PipelineNamespaceFilter,
+        hydrate: bool = False,
+    ) -> Page[PipelineNamespaceResponse]:
+        """List all pipeline namespaces matching the given filter criteria.
+
+        Args:
+            filter_model: All filter parameters including pagination
+                params.
+            hydrate: Flag deciding whether to hydrate the output model(s)
+                by including metadata fields in the response.
+
+        Returns:
+            A list of all pipeline namespaces matching the filter criteria.
+        """
+
+        def _custom_conversion(
+            name: str,
+        ) -> PipelineNamespaceResponse:
+            runs = self.list_runs(
```

Comment:
This current implementation of getting the ID/status of the latest run is inefficient as it runs another DB query for each pipeline namespace separately, however I'm not sure how to best implement this otherwise.

My main `filter_and_paginate` call only queries the `PipelineSchema.name` in order to use the `.distinct()` option so we can re-use the filter/pagination code.

@AlexejPenner any idea maybe?

Reply:
ah, i missed something, let me rewrite